### PR TITLE
Adds types

### PR DESCRIPTION
**Pull request by dbrudner**

### CODE REVIEW

File: types/test-types.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  setPricing(checkoutPricingInput);
+  // $ExpectError
+  setPricing();
+  // $ExpectError
+  setPricing({ ...checkoutPricingInput, fake: "fake" });
+
+  // $ExpectType boolean
+  loading;
+  // $ExpectType string
+  price.now.items[0].addons;
```
Comment (dbrudner) [inline types/test-types.tsx:176]: There might be a better way to make these assertions. We're already testing this time in [recurly-js](https://github.com/recurly/recurly-js/blob/master/test/types/pricing/checkout.ts#L103-L164)

File: package.json [hunk 2]
```diff
@@ -18,6 +19,8 @@
     "react-dom": "^16.8.0-0"
   },
   "dependencies": {
+    "@types/react": "^16.9.32",
```
Comment (dbrudner) [inline package.json:22]: These need to be included as dependencies https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+export type UseCheckoutPricingState = {
+  price: CheckoutPrice;
+  loading: boolean;
+};
+
+export type SetCheckoutPricing = (input: UseCheckoutPricingInput) => void;
+
+export type UseCheckoutPricingReturn = [UseCheckoutPricingState, SetCheckoutPricing];
+
+/**
```
Comment (dbrudner) [inline types/index.d.ts:104]: The tsdoc comments show like this 
image

File: types/test-types.tsx (hunk 1 above)
Comment (chrissrogers) [reply types/test-types.tsx:176]: Seems reasonable since the cause differs

Comment (dbrudner) [reply types/test-types.tsx:176]: That's what I was thinking, but I wasn't sure

File: .npmignore [hunk 4]
```diff
@@ -10,3 +10,7 @@ Makefile
 webpack.config.js
 docs
 README.mdx
+
+types/test.tsx
```
Comment (dbrudner) [inline .npmignore:14]: this should be types/test-types.tsx

**Pull request merged**